buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/NamingConventionsTask.java

```diff
@@ -30,16 +32,25 @@ public NamingConventionsTask() {
         final Project project = getProject();
 
         SourceSetContainer sourceSets = getJavaSourceSets();
-        final FileCollection classpath = project.files(
-                // This works because the class only depends on one class from junit that will be available from the
-                // tests compile classpath. It's the most straight forward way of telling Java where to find the main
-                // class.
-                NamingConventionsCheck.class.getProtectionDomain().getCodeSource().getLocation().getPath(),
-                // the tests to be loaded
-                checkForTestsInMain ? sourceSets.getByName("main").getRuntimeClasspath() : project.files(),
-                sourceSets.getByName("test").getCompileClasspath(),
-                sourceSets.getByName("test").getOutput()
-        );
+        final FileCollection classpath;
+        try {
+            URL location = NamingConventionsCheck.class.getProtectionDomain().getCodeSource().getLocation();
+            if (location.getProtocol().equals("file") == false) {
+                throw new GradleException("Unexpected location for NamingConventionCheck class: "+ location);
+            }
+            classpath = project.files(
+                    // This works because the class only depends on one class from junit that will be available from the
+                    // tests compile classpath. It's the most straight forward way of telling Java where to find the main
+                    // class.
+                    location.toURI().getPath(),
+                    // the tests to be loaded
+                    checkForTestsInMain ? sourceSets.getByName("main").getRuntimeClasspath() : project.files(),
+                    sourceSets.getByName("test").getCompileClasspath(),
+                    sourceSets.getByName("test").getOutput()
+            );
+        } catch (URISyntaxException e) {
+            throw new GradleException("Failed to find NamingConventionsCheck class", e);
```

Comment:
I think this message is a little deceptive. The class can be found, but somehow the protection domain does not have a valid URL. This should be an AssertionError (should be impossible to hit).